c10/core/SymInt.h

```diff
@@ -63,7 +58,22 @@ class C10_API SymInt {
   }
 
  private:
-  const static int64_t SYM_TAG_MASK = 1LL << 63;
+  // Constraints on the internal representation:
+  // - Should represent positive and negative ints
+  // - No conversion necessary for operations on ints.
+  // - We reserve some values to act as indices into our sym int table.
+  //
+  // So, the scheme is to reserve large negative numbers:
+  // - 0b11... means we are a negative int (following two's complement)
+  // - 0b00... means we are a positive int (following two's complement)
```

Comment:
nit: INT64_MAX is 0b0111111...., so positive numbers are both 0b00xxx and 0b01xxx?

Reply:
rewrote the comment!